src/main/java/jenkins/plugins/logstash/LogstashConfiguration.java

```diff
@@ -97,13 +102,21 @@ public void migrateData()
             break;
           case ELASTICSEARCH:
             LOGGER.log(Level.INFO, "Migrating logstash configuration for Elastic Search");
-            ElasticSearch es = new ElasticSearch();
-            es.setHost(descriptor.getHost());
-            es.setPort(descriptor.getPort());
-            es.setKey(descriptor.getKey());
-            es.setUsername(descriptor.getUsername());
-            es.setPassword(descriptor.getPassword());
-            logstashIndexer = es;
+            URI uri;
+            try
+            {
+              uri = (new URIBuilder(descriptor.getHost()))
+                  .setPort(descriptor.getPort())
+                  .setPath("/" + descriptor.getKey()).build();
```

Comment:
not sure if this will work properly for keys that already start with `/`
does this `UriBuilder` collapse consecutive `/`?

Reply:
It does not and this is exactly the same behaviour as it was before in the ElasticSearchDao

Reply:
OK then